viewer/packages/camera-manager/src/CameraManager.ts

```diff
@@ -1,604 +1,58 @@
-        ? Math.min(this._camera.position.distanceTo(modelsBoundingBox.getCenter(new THREE.Vector3())), modelSize) / 2
-        : lastScrollTargetDistance;
-
-    this._raycaster.setFromCamera(cursorPosition, this._camera);
-
-    const farPoint = this._raycaster.ray.direction
-      .clone()
-      .normalize()
-      .multiplyScalar(newTargetDistance)
-      .add(this._camera.position);
-
-    return farPoint;
-  }
+  getCameraState(): Required<CameraState>;
 
   /**
-   * Calculates new camera target based on cursor position.
-   * @param event MouseEvent that contains pointer location data.
+   * Required for understadning of `cameraChanged` definition for other parts of Reveal
+   * @param event
+   * @param callback
    */
-  private async calculateNewTarget(event: MouseEvent): Promise<THREE.Vector3> {
-    const { offsetX, offsetY } = event;
-    const { x, y } = this.convertPixelCoordinatesToNormalized(offsetX, offsetY);
-
-    const modelRaycastData = await this._modelRaycastCallback(offsetX, offsetY);
-
-    const newTarget =
-      modelRaycastData.intersection?.point ??
-      this.calculateNewTargetWithoutModel({ x, y }, modelRaycastData.modelsBoundingBox);
-
-    return newTarget;
-  }
-
+  on(event: 'cameraChange', callback: CameraChangeData): void;
   /**
-   * Removes controls event listeners if they are defined.
+   * Required for understadning of `cameraChanged` definition for other parts of Reveal
+   * @param event
+   * @param callback
    */
-  private teardownControls(removeOnWheel: boolean = true): void {
-    if (this._onClick !== undefined) {
-      this._inputHandler.off('click', this._onClick as PointerEventDelegate);
-      this._onClick = undefined;
-    }
-    if (this._onWheel !== undefined && removeOnWheel) {
-      this._domElement.removeEventListener('wheel', this._onWheel);
-      this._onWheel = undefined;
-    }
-  }
-
-  private handleMouseWheelActionChange(controlsOptions: CameraControlsOptions) {
-    const { _controls: controls } = this;
+  off(event: 'cameraChange', callback: CameraChangeData): void;
 
-    switch (controlsOptions?.mouseWheelAction) {
-      case 'zoomToTarget':
-        controls.zoomToCursor = false;
-        break;
-      case 'zoomPastCursor':
-        controls.useScrollTarget = false;
-        controls.zoomToCursor = true;
-        break;
-      case 'zoomToCursor':
-        controls.setScrollTarget(controls.getState().target);
-        controls.useScrollTarget = true;
-        controls.zoomToCursor = true;
-        break;
-      case undefined:
-        break;
-      default:
-        assertNever(controlsOptions.mouseWheelAction);
-    }
-  }
+  fitCameraToBoundingBox(boundingBox: THREE.Box3, duration?: number, radiusFactor?: number): void;
   /**
-   * Method for setting up camera controls listeners and values inside current controls class.
+   * Updates internal state of camera manager.
+   * @param deltaTime
+   * @param boundingBox
    */
```

Comment:
This doesn't tell anything to implementors about what they should do in their implementations. This description must be detailed and explain in detail what the function is expected to do and what the arguments are.